tests/mock_network/messages.rs

```diff
@@ -90,14 +91,23 @@ fn send_and_receive() {
 
     for node in nodes
         .iter_mut()
-        .filter(|n| n.inner.is_elder() && n.is_recipient(&dst))
+        .filter(|n| n.inner.is_elder() && n.in_dst_location(&dst))
     {
         loop {
             match node.try_next_ev() {
-                Ok(Event::MessageReceived { content, src, dst }) => {
+                Ok(Event::MessageReceived { content, src, .. }) => {
                     request_received_count += 1;
                     if req_content == content {
-                        if let Err(err) = node.inner.send_message(dst, src, res_content.clone()) {
+                        let res_src = SrcLocation::PrefixSection(*node.our_prefix());
```

Comment:
This change look a bit out of place in this commit (Splitting Location into SrcLocation and DstLocation, why is it relevant  here. I'm also not clear what we are trying to do here.

Reply:
Previously we would just flip the src and dst when sending the response, but we can't do that anymore as they are different types. Also in later commit they actually have different variants so a simple mapping of src <-> dst won't work.

Reply:
Wouldn't it make sense to do it in the later commit then where it is necessary?
It just seem odd since we never use PrefixSection as a source so far (I think) (in later commit we start using the Prefix as the source, but I think we make it behave differently).

Reply:
> Wouldn't it make sense to do it in the later commit then where it is necessary

But it is necessary here. I can't just take `SrcLocation` and pass it to where `DstLocation` is expected (which is what we did previously because they were the same types). Or am I misunderstanding what you mean?

Reply:
I mean you can match the dst to a DstLocation::Section(name) => SrcLocation(name) no?

Reply:
Like you do for the res_dst.